specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json

```diff
@@ -0,0 +1,2402 @@
+    "version": "2023-02-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Elastic/operations": {
+      "get": {
+        "tags": [
+          "MonitorsResource"
+        ],
+        "summary": "List all operations provided by Microsoft.Elastic.",
+        "operationId": "Operations_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderDefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Operations_List": {
+            "$ref": "./examples/Operations_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Elastic/monitors": {
+      "get": {
+        "tags": [
+          "MonitorsResource"
+        ],
+        "summary": "List all monitors under the specified subscription.",
+        "operationId": "Monitors_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ElasticMonitorResourceListResponse"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderDefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Monitors_List": {
+            "$ref": "./examples/Monitors_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Elastic/monitors/getOrganizationApiKey/{emailId}": {
```

Comment:
>/monitors/getOrganizationApiKey/{emailId}" [](http://example.com/codeflow?start=64&length=42)

this violates the ARM API guidelines for segments

Reply:
POST actions MUST have {actionName} which is a verb as the last segment. Also after the namespace name the path MUST have {resourceType}/{resourceTypoeName} tuples.

Reply:
@rkmanda

Pushed new changes in https://github.com/Azure/azure-rest-api-specs/pull/22638/commits/7d202dde038327e0c866c0b56baf2697eaf43fb5.

Doubt: If we give the path as "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Elastic/monitors/{monitorName}/getOrganizationApiKey", what is the behavior if the monitor resource is deleted?

The concern is because we want the "getOrganizationApiKey" operation to be independent of the monitor resource.

Could you suggest any alternatives?

Reply:
If the action is not related to a resource it should be POST .../Microsoft.Elastic/getOrganizationApiKey

Reply:
Fixed as part of https://github.com/Azure/azure-rest-api-specs/pull/22638/commits/365ec886461cfe9429f7c574bc15995d47f5e88b